flow/blocks/filters/ema.py

```diff
@@ -19,9 +21,21 @@ def compute(self, inputs, params):
             period_history_length = period_history_length - 1
         self.period_history.append(result)
 
-        last_average = result
-        alpha = 2.0 / (moving_average_periods + 1.0)
-        for item in self.period_history:
-            last_average = (item * alpha) + (last_average * (1.0-alpha))
+        if (self.last_average is not None):
+            if(result is not None):
+                alpha = 2.0 / (moving_average_periods + 1.0)
+                new_average = result * alpha + self.last_average * (1 - alpha)
+            else:
+                new_average = self.last_average
+        else:
+            if(result is not None):
+                new_average = result
 
-        return last_average
+        self.last_average = new_average
+
+        # last_average = result
```

Comment:
Did you mean to leave this commented code in?

Reply:
Yeah, we've had some back-and-forth about the proper ema calculation (both flow-server and flow client had odd/non-standard calculations) so we left the previous methods in as comments in the rare case that someone wants to refer to it.